aten/src/ATen/native/cuda/Reduce.cuh

```diff
@@ -593,42 +707,24 @@ inline void gpu_reduce_kernel(TensorIterator& iter, const ops_t& ops, ident_t id
     AT_CUDA_CHECK(cudaMemsetAsync(semaphores.get(), 0, config.semaphore_size(), stream));
   }
 
-  if (can_use_32bit_indexing) {
-    auto output_calc = make_output_calculator<uint32_t>(iter);
-    auto input_calc = make_input_calculator<uint32_t>(iter);
-    auto reduce = ReduceOp<scalar_t, ops_t, uint32_t, out_scalar_t>(
-        ops,
-        config,
-        input_calc,
-        output_calc,
-        in_data,
-        out_data,
-        buffer.get(),
-        (int*)semaphores.get(),
-        ident);
-    reduce.accumulate = iter.should_accumulate();
-    reduce.final_output = iter.is_final_output();
-
-    launch_reduce_kernel<ReduceConfig::MAX_NUM_THREADS>(config, reduce);
-  } else {
-    auto output_calc = make_output_calculator<uint64_t>(iter);
-    auto input_calc = make_input_calculator<uint64_t>(iter);
-    auto reduce = ReduceOp<scalar_t, ops_t, uint64_t, out_scalar_t>(
-        ops,
-        config,
-        input_calc,
-        output_calc,
-        in_data,
-        out_data,
-        buffer.get(),
-        (int*)semaphores.get(),
-        ident);
-    AT_ASSERT(!iter.should_accumulate());
-    reduce.accumulate = false;
-    reduce.final_output = true;
-
-    launch_reduce_kernel<ReduceConfig::MAX_NUM_THREADS>(config, reduce);
-  }
+  assert(can_use_32bit_indexing);
```

Comment:
Please don't use `assert`. `AT_ASSERT` is what you want.

Reply:
Will update